core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,44 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        $scope.userIsLoggedIn = GLOBALS.userIsLoggedIn;
+        $scope.activitiesOwned = {explorations: {}, collections: {}};
+        if ($scope.userIsLoggedIn) {
+          $http.get('/creatordashboardhandler/data')
+            .then(function(response) {
+              $scope.libraryGroups.forEach(function(activityGroups) {
+                var activitySummaryDict = activityGroups.activity_summary_dicts;
+
+                var ACTIVITY_TYPE_EXPLORATION = 'exploration';
+                var ACTIVITY_TYPE_COLLECTION = 'collection';
+                activitySummaryDict.forEach(function(activity) {
```

Comment:
The dummy var should be activitySummaryDict, right?

Please be very careful to figure out exactly what each variable means. The naming conventions that you've used make this code really hard to follow.

Reply:
Same thing here. The first thing that confused me was that activitySummaryDict had a forEach --
that's not the way to iterate over a dict. And then on further investigation it seemed like it wasn't a dict at all.

So, again, this is a good time to apply the `things.forEach(function(thing)` principle I talked about above.